backend/src/main/java/vn/edu/hust/nmcnpm_20242_n3/controller/UserController.java

```diff
@@ -105,35 +72,18 @@ public ResponseEntity<?> updateUser(@PathVariable String id, @RequestBody UserDT
         }
     }
     @DeleteMapping("/{id}")
-    public ResponseEntity<?> deleteUser(@PathVariable String id, @RequestParam String currentUserId) {
-        User currentUser = userService.getUserEntityById(currentUserId);
-        if (!permissionService.hasPermission(currentUser, PermissionEnum.MANAGE_USERS)) {
-            return ResponseEntity.status(403).body("Forbidden: No permission");
-        }
-
+    public ResponseEntity<?> deleteUser(@PathVariable String id) {
         userService.deleteUser(id);
         return ResponseEntity.ok("User deleted");
     }
     @GetMapping("/{userId}/book-loans")
-    public ResponseEntity<?> getBookLoansByUserId(@PathVariable String userId, @RequestParam String currentUserId) {
-        User currentUser = userService.getUserEntityById(currentUserId);
-        if (!permissionService.hasPermission(currentUser, PermissionEnum.MANAGE_USERS)
-                && !currentUser.getId().equals(userId)) {
-            return ResponseEntity.status(403).body("Forbidden: No permission");
-        }
-
+    public ResponseEntity<?> getBookLoansByUserId(@PathVariable String userId) {
         List<BookLoan> loans = userService.getBookLoansByUserId(userId);
         return ResponseEntity.ok(loans);
     }
 
     @GetMapping("/{userId}/fines")
-    public ResponseEntity<?> getFinesByUserId(@PathVariable String userId, @RequestParam String currentUserId) {
-        User currentUser = userService.getUserEntityById(currentUserId);
-        if (!permissionService.hasPermission(currentUser, PermissionEnum.MANAGE_USERS)
-                && !currentUser.getId().equals(userId)) {
-            return ResponseEntity.status(403).body("Forbidden: No permission");
-        }
-
+    public ResponseEntity<?> getFinesByUserId(@PathVariable String userId) {
         List<Fine> fines = userService.getFinesByUserId(userId);
         return ResponseEntity.ok(fines);
     }
```

Comment:
I think that we will need a `FineDTO` and `BookLoanDTO` here, as for now, this will return an infinite recursive reference between `BookLoan` and `User`